sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,358 @@
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+
+@Fluent
+public final class CloudEvent {
+    private static final String SPEC_VERSION = "1.0";
+    private static final JsonSerializer DESERIALIZER = JsonSerializerProviders.createInstance();
+    private static final ClientLogger LOGGER = new ClientLogger(CloudEvent.class);
+    private static final Set<String> RESERVED_ATTRIBUTE_NAMES = new HashSet<>(Arrays.asList(
+        "specversion",
+        "id",
+        "source",
+        "type",
+        "datacontenttype",
+        "dataschema",
+        "subject",
+        "time",
+        "data"
+    ));
+
+    /*
+     * An identifier for the event. The combination of id and source must be
+     * unique for each distinct event.
+     */
+    @JsonProperty(value = "id", required = true)
+    private String id;
+
+    /*
+     * Identifies the context in which an event happened. The combination of id
+     * and source must be unique for each distinct event.
+     */
+    @JsonProperty(value = "source", required = true)
+    private String source;
+
+    /*
+     * Event data specific to the event type.
+     */
+    @JsonProperty(value = "data")
+    private Object data;
+
+    /*
+     * Event data specific to the event type, encoded as a base64 string.
+     */
+    @JsonProperty(value = "data_base64")
+    private byte[] dataBase64;
+
+    /*
+     * Type of event related to the originating occurrence.
+     */
+    @JsonProperty(value = "type", required = true)
+    private String type;
+
+    /*
+     * The time (in UTC) the event was generated, in RFC3339 format.
+     */
+    @JsonProperty(value = "time")
+    private OffsetDateTime time;
+
+    /*
+     * The version of the CloudEvents specification which the event uses.
+     */
+    @JsonProperty(value = "specversion", required = true)
+    private String specVersion;
+
+    /*
+     * Identifies the schema that data adheres to.
+     */
+    @JsonProperty(value = "dataschema")
+    private String dataSchema;
+
+    /*
+     * Content type of data value.
+     */
+    @JsonProperty(value = "datacontenttype")
+    private String dataContentType;
+
+    /*
+     * This describes the subject of the event in the context of the event
+     * producer (identified by source).
+     */
+    @JsonProperty(value = "subject")
+    private String subject;
+
+    @JsonIgnore
+    private Map<String, Object> extensionAttributes;
+
+    /*
+     * Cache serialized data for getData()
+     */
+    @JsonIgnore
+    private BinaryData binaryData;
+
+    private CloudEvent() {
+
+    }
+
+    public CloudEvent(String source, String type, Object data) {
+        this(source, type, data, null);
+    }
+
+    public CloudEvent(String source, String type, Object data, String dataContentType) {
+        this.source = source;
```

Comment:
Line 158 check for mandatory fields .  If any of these parameters are mandatory, shouldn't we check at creating time  `Objects.requireNonNull()` ?

Reply:
Added mandatory check
```java
        if (Objects.isNull(source)) {
            throw LOGGER.logExceptionAsError(new NullPointerException("'source' cannot be null."));
        }
        if (Objects.isNull(type)) {
            throw LOGGER.logExceptionAsError(new NullPointerException("'type' cannot be null."));
        }
        if (Objects.isNull(data)) {
            throw LOGGER.logExceptionAsError(new NullPointerException("'data' cannot be null."));
        }
        if (Objects.isNull(format)) {
            throw LOGGER.logExceptionAsError(new NullPointerException("'format' cannot be null."));
        }
```

Reply:
Thanks for the suggestion. Changed to requiresNonNull.